### PR TITLE
Ability to set up an Agent in Query database action mode

### PR DESCRIPTION
**Pull request by PopDaph**
### Context

This adds a new "Query database" action in the Assistant builder. Only for Dust and local workspaces for now

Saving creates an AgentConfiguration with a reference to a new `AgentDatabaseConfiguration` model > we store the id of the datasource and the id of the database.

This PR does not handle the run part so for now those Assistants are useless; this is for the sake of not submitting a huge PR. 

### Deployment

Requires running init_db on Front.

### Screenshots

When selecting the Action "Query Database": 
Screenshot 2023-12-11 at 13 04 08

Modal to select a database: 
Screenshot 2023-12-11 at 15 25 36
Screenshot 2023-12-11 at 15 25 40

Screenshot 2023-12-11 at 15 25 49

Once database is selected:

### CODE REVIEW

File: front/lib/models/assistant/actions/database.ts [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+  InferCreationAttributes,
+  Model,
+} from "sequelize";
+
+import { front_sequelize } from "@app/lib/databases";
+
+/**
+ * Action Database Configuration
+ */
+export class AgentDatabaseConfiguration extends Model<
```
Comment (spolu) [inline front/lib/models/assistant/actions/database.ts:14]: Let's call it `AgentDatabaseQueryConfiguration` maybe to continue with the convention that these things have an action "verb" in their name?

File: types/src/front/assistant/actions/database.ts [hunk 2]
```diff
@@ -0,0 +1,30 @@
+import { ModelId } from "shared/model_id";
+
+import { AgentActionConfigurationType } from "../../../front/assistant/agent";
+import { AgentActionType } from "../../../front/assistant/conversation";
+
+export type DatabaseConfigurationType = {
+  id: ModelId;
+  sId: string;
+  type: "database_configuration";
```
Comment (spolu) [inline types/src/front/assistant/actions/database.ts:9]: `database_query_configuration`

File: types/src/front/assistant/actions/database.ts [hunk 3]
```diff
@@ -0,0 +1,30 @@
+import { ModelId } from "shared/model_id";
+
+import { AgentActionConfigurationType } from "../../../front/assistant/agent";
+import { AgentActionType } from "../../../front/assistant/conversation";
+
+export type DatabaseConfigurationType = {
```
Comment (spolu) [inline types/src/front/assistant/actions/database.ts:6]: Let's call it `DatabaseQueryConfigurationType`

File: types/src/front/assistant/actions/database.ts [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+  id: ModelId;
+  sId: string;
+  type: "database_configuration";
+  dataSourceId: string;
```
Comment (spolu) [inline types/src/front/assistant/actions/database.ts:10]: You want a `dataSourceWorkspaceId` here

File: types/src/front/api_handlers/internal/agent_configuration.ts [hunk 5]
```diff
@@ -74,6 +74,11 @@ export const PostOrPatchAgentConfigurationRequestBodySchema = t.type({
         appWorkspaceId: t.string,
         appId: t.string,
       }),
+      t.type({
+        type: t.literal("database_configuration"),
```
Comment (spolu) [inline types/src/front/api_handlers/internal/agent_configuration.ts:78]: You want a `dataSourceWorkspaceId` here

File: front/components/assistant_builder/AssistantBuilder.tsx [hunk 6]
```diff
@@ -115,6 +120,14 @@ export type AssistantBuilderDustAppConfiguration = {
   app: AppType;
 };
 
+// Database Action
+
+export type AssistantBuilderDatabaseConfiguration = {
```
Comment (spolu) [inline front/components/assistant_builder/AssistantBuilder.tsx:125]: I would add a Query here as well :grimacing:

**Review by spolu — COMMENTED** [review]
Left a few naming comments + change request on model/type (we need to capture the dataSourceWorkspaceId)

I think it's really worth having a DataSource then Database selection as the current view will not scale as is and this will require a rather painful refactor

**Review by fontanierh — DISMISSED** [review]
Code looks great!

**Comment by PopDaph** [comment]
@spolu feedback applied! 🙇🏻‍♀️

File: front/lib/models/assistant/actions/database.ts [hunk 7]
```diff
@@ -0,0 +1,73 @@
...
+> {
+  declare id: CreationOptional<number>;
+  declare createdAt: CreationOptional<Date>;
+  declare updatedAt: CreationOptional<Date>;
+
+  declare sId: string;
+
+  declare dataSourceId: string;
+  declare databaseId: string;
+  declare appWorkspaceId: string;
```
Comment (spolu) [inline front/lib/models/assistant/actions/database.ts:26]: Same here of course 👍

**Review by spolu — COMMENTED** [review]
Looks great one final change request, feel free to merge once addressed 👍

File: front/components/assistant_builder/AssistantBuilder.tsx [hunk 8]
```diff
@@ -115,6 +120,15 @@ export type AssistantBuilderDustAppConfiguration = {
   app: AppType;
 };
 
+// Database Query Action
+
+export type AssistantBuilderDatabaseQueryConfiguration = {
+  dataSourceId: string;
+  databaseId: string;
+  databaseName: string;
+  appWorkspaceId: string;
```
[the remark this answers is not in the document]
Comment (PopDaph) [reply front/components/assistant_builder/AssistantBuilder.tsx:129]: 🤦🏻‍♀️  adding a commit!

**Pull request merged**